modules/dnn/src/cuda4dnn/primitives/pooling.hpp

```diff
@@ -0,0 +1,258 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+
+#ifndef OPENCV_DNN_SRC_CUDA4DNN_PRIMITIVES_POOLING_HPP
+#define OPENCV_DNN_SRC_CUDA4DNN_PRIMITIVES_POOLING_HPP
+
+#include "../../op_cuda.hpp"
+
+#include "../csl/cudnn.hpp"
+#include "../csl/tensor.hpp"
+#include "../csl/tensor_ops.hpp"
+
+#include <opencv2/core.hpp>
+
+#include <cstddef>
+#include <cstdint>
+#include <vector>
+#include <utility>
+#include <algorithm>
+
+namespace cv { namespace dnn { namespace cuda4dnn {
+
+    struct PoolingConfiguration {
+        enum class pooling_mode {
+            max,
+            average_included, /* include padding while calculating average */
+            average_excluded /* exclude padding while calculating average */
```

Comment:
It would be nice to follow coding style:
- constants are UPPER_CASE
- struct/types are CamelCase (except simple scalars or with `_t` suffix)
- variables/fields are lower_case_with_underscores
- `methodNames()`

Reply:
I was originally using full caps but decided to use small caps since the constants declared in `enum class` (not `enum`) are qualified with the enum name (`pooling_mode::max`).

Should I use an `enum` or `enum class`?

Reply:
enum name should be in CamelCase (like a type name) - `PoolingMode::MAX`

> `enum` or `enum class`

`enum class` is acceptable.